test/fixtures/rules/std_rule_cases/AM08.yml

```diff
@@ -0,0 +1,140 @@
+rule: AM08
+
+test_pass_select_comma:
+  # this syntax is currently not covered by AM08
+  pass_str: "SELECT a.*, b.* FROM a, b"
+
+test_fail_missing_clause:
+  fail_str: "SELECT foo.a, bar.b FROM foo INNER JOIN bar;\n"
+  fix_str: "SELECT foo.a, bar.b FROM foo CROSS JOIN bar;\n"
+
+test_fail_missing_clause_lowercase_default:
+  fail_str: "SELECT foo.a, bar.b FROM foo left join bar;\n"
+  fix_str: "SELECT foo.a, bar.b FROM foo cross join bar;\n"
+
+test_fail_missing_multiple_joins:
+  fail_str: "SELECT foo.a, bar.b FROM foo left join bar on 1=2 left join baz;\n"
+  fix_str: "SELECT foo.a, bar.b FROM foo left join bar on 1=2 cross join baz;\n"
+
+test_fail_missing_multiple_joins_mixed_types:
+  fail_str: "SELECT foo.a, bar.b FROM foo left join bar on foo.x = bar.y left join baz;\n"
+  fix_str: "SELECT foo.a, bar.b FROM foo left join bar on foo.x = bar.y cross join baz;\n"
+
+test_fail_missing_multiple_joins_not_last:
+  fail_str: "SELECT foo.a, bar.b FROM foo left join bar left join baz on foo.x = bar.y;\n"
+  fix_str: "SELECT foo.a, bar.b FROM foo cross join bar left join baz on foo.x = bar.y;\n"
+
+test_fail_cross_join_with_where_clause:
+  fail_str: "SELECT foo.a, bar.b FROM foo JOIN bar WHERE foo.a = bar.a;\n"
+  violations:
+    - code: AM08
+      description: WHERE clause used for join condition. Use explicit ON instead.
+      end_file_pos: 57
+      end_line_no: 1
+      end_line_pos: 58
+      fixes: []
+      name: ambiguous.join_condition
+      start_file_pos: 38
+      start_line_no: 1
+      start_line_pos: 39
+      warning: false
+
+test_fail_cross_join_with_complex_where_clause:
+  fail_str: "SELECT foo.a, bar.b FROM foo JOIN bar WHERE foo.a = bar.a AND foo.x = 3;\n"
+  violations:
+    - code: AM08
+      description: WHERE clause used for join condition. Use explicit ON instead.
+      end_file_pos: 71
+      end_line_no: 1
+      end_line_pos: 72
+      fixes: []
+      name: ambiguous.join_condition
+      start_file_pos: 38
+      start_line_no: 1
+      start_line_pos: 39
+      warning: false
+
+test_pass_cross_join_with_complex_where_clause_not_simplifiable:
+  pass_str: "SELECT foo.a, bar.b FROM foo JOIN bar WHERE foo.a = bar.a OR foo.x = 3;\n"
+
+test_pass_cross_join_with_complex_where_clause_and_subqueries:
+  fail_str: "SELECT foo.a, bar.b FROM (SELECT * FROM abc) foo JOIN (SELECT * FROM def WHERE x = 3) bar WHERE foo.a = bar.a;\n"
+
+test_pass_inner_join_default:
+  pass_str: "SELECT foo.a, bar.b FROM foo INNER JOIN bar ON 1=1;\n"
+
+test_pass_left_join_default:
+  pass_str: "SELECT foo.a, bar.b FROM foo LEFT JOIN bar ON 1=1;\n"
```

Comment:
Personally I think this is yet another different rule maybe.
`fixed_bool_condition_style`: `TRUE/FALSE, 1 = 1/ =0, never`

Personally I hate this style cause its redundant code, and i find it funny that users think they need to evaluate something to get a condition to be TRUE/FALSE. However sadly its also not possible to remove cause people often use it so that all other condition are `AND $cond`

Reply:
I am happy to create backlog issue but I'd prefer to avoid feature creep in this PR.

Reply:
https://github.com/sqlfluff/sqlfluff/issues/6350